lib/haml/railtie.rb

```diff
@@ -10,13 +10,32 @@
 module Haml
   class Railtie < ::Rails::Railtie
     initializer :haml do |app|
-      require "haml/template"
-      if defined?(::Sass::Rails::SassTemplate) && app.config.assets.enabled
-        require "haml/sass_rails_filter"
+      if defined?(ActiveSupport)
+        ActiveSupport.on_load(:before_initialize) do
```

Comment:
I think the 'on_load(:action_view) is the important bit. But if we can get rid of the `defined?(ActiveSupport)` then a lot can be simplified.